src/components/legend/draw.js

```diff
@@ -30,26 +30,44 @@ var getLegendData = require('./get_legend_data');
 var style = require('./style');
 var helpers = require('./helpers');
 
-module.exports = function draw(gd) {
+module.exports = function draw(gd, opts) {
     var fullLayout = gd._fullLayout;
     var clipId = 'legend' + fullLayout._uid;
+    var layer;
 
-    if(!fullLayout._infolayer || !gd.calcdata) return;
+    // Check whether this is the main legend (ie. called without any opts)
+    if(!opts) {
```

Comment:
IMO, if this is really an option, on top of the function we should be able to add
```js
if(!opts) opts = {}:
```
But we can't do that right now!
So this line should be something like
```js
if(!opts._unifiedHover) {
```
and we must pass `_unifiedHover` in unified-hover `opts` object.

Reply:
I'm not sure I see why this
```
if(!opts) opts = {};
```
is much better than:
```
if(!opts) {
  opts = fullLayout.legend || {};
  ... some more defaults values ...
}
```
Can you show me a commit with the changes you'd like to see @archmoj ?

Reply:
See https://github.com/plotly/plotly.js/commit/72abd2fb0f72344b13065319c66d0211094fb73f